examples/usage/cooldown.rs

```diff
@@ -0,0 +1,139 @@
+//! This example demonstrates how you can implement a cooldown in UI.
+//! We create three buttons with 2, 1, and 5 seconds cooldown.
+
+use bevy::{
+    color::palettes::tailwind::{SLATE_400, SLATE_50},
+    ecs::spawn::SpawnIter,
+    prelude::*,
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, (activate_ability, animate_cooldowns))
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    mut texture_atlas_layouts: ResMut<Assets<TextureAtlasLayout>>,
+    asset_server: Res<AssetServer>,
+) {
+    commands.spawn(Camera2d);
+    let texture = asset_server.load("textures/food_kenney.png");
+    let layout = TextureAtlasLayout::from_grid(UVec2::splat(64), 7, 7, None, None);
+    let texture_atlas_layout = texture_atlas_layouts.add(layout);
+    commands.spawn((
+        Node {
+            width: Val::Percent(100.),
+            height: Val::Percent(100.),
+            align_items: AlignItems::Center,
+            justify_content: JustifyContent::Center,
+            column_gap: Val::Px(15.),
+            ..default()
+        },
+        Children::spawn(SpawnIter(
+            [
+                ("an apple", 2., 2),
+                ("a burger", 1., 23),
+                ("chocolate", 10., 32),
+                ("cherries", 4., 41),
+            ]
+            .into_iter()
+            .map(move |ability| {
+                build_ability(ability, texture.clone(), texture_atlas_layout.clone())
+            }),
+        )),
+    ));
+    commands.spawn((
+        Text::new("*Click some foot to eat it*"),
+        Node {
+            position_type: PositionType::Absolute,
+            top: Val::Px(5.0),
+            left: Val::Px(15.0),
+            ..default()
+        },
+    ));
+}
+
+type Ability = (&'static str, f32, usize);
+
+fn build_ability(
+    ability: Ability,
+    texture: Handle<Image>,
+    layout: Handle<TextureAtlasLayout>,
+) -> impl Bundle {
+    let (name, cooldown, index) = ability;
+    let name = Name::new(name);
+
+    (
+        Node {
```

Comment:
We should have a comment here to explain the approach we use to visualize the cooldown. Briefly explain that the "clip overflow" + child entity will accomplish the effect of visually "filling up" the UI node.